specs/hotel_spec.rb

```diff
@@ -0,0 +1,147 @@
+require_relative 'spec_helper'
+
+describe "Hotel class" do
+
+  describe "Instantiation" do
+
+    it "Can be instantiated" do
+      Hotel::Hotel.new(20, 200).must_be_instance_of Hotel::Hotel
+    end
+
+    it "Can create an array of rooms" do
+      Hotel::Hotel.new(20, 200).rooms.length.must_equal 20
+    end
+
+    it "Can call the array of rooms" do
+      Hotel::Hotel.new(20,200).rooms.must_equal [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18, 19, 20]
+    end
+
+    it "Can call on price of rooms" do
+      Hotel::Hotel.new(20, 200).price.must_equal 200
+    end
+
+    it "can call on collection of reservations array - will be empty" do
+      Hotel::Hotel.new(20, 200).reservation_collection.must_be_instance_of Array
+      Hotel::Hotel.new(20, 200).block_reservation_collection.must_be_instance_of Array
+
+    end
+  end #describe instantiate
+
+  describe "#make reservation" do
+    before do
+      @bb_hotel = Hotel::Hotel.new(4, 200)
+      @bb_hotel.make_reservation('sept 8 2017', 'sept 10 2017', 4)
+    end
+
+    it "can be instantiated" do
+      @bb_hotel.reservation_made.reservation_array.must_include 4
+      @bb_hotel.reservation_collection.length.must_equal 1
+    end
+
+    it "Can call on class reservation methods" do
+      @bb_hotel.reservation_made.room_num.must_equal 4
+    end
+
+    it "Can return an argument error if room is not available for that date range" do
+      proc { @bb_hotel.make_reservation('sept 8 2017', 'sept 9 2017', 4)}.must_raise ArgumentError
+    end  ####NOTE##### Not sure how to create a test to check if there is no problem with the code
```

Comment:
Checking for an exception is a good call here. If you're trying to avoid it passing when you've written a bug, a common technique is to define your own exception and look for that instead.